src/sentry/scim/endpoints/teams.py

```diff
@@ -0,0 +1,213 @@
+
+        with transaction.atomic():
+            omt = OrganizationMemberTeam.objects.get(team=team, organizationmember=member)
+            self.create_audit_entry(
+                request=request,
+                organization=team.organization,
+                target_object=omt.id,
+                target_user=member.user,
+                event=AuditLogEntryEvent.MEMBER_LEAVE_TEAM,
+                data=omt.get_audit_log_data(),
+            )
+            omt.delete()
+
+    def _rename_team_opereation(self, request, operation, team):
+        serializer = TeamSerializer(
+            team,
+            data={
+                "slug": slugify(operation["value"]["displayName"]),
+            },
+            partial=True,
+        )
+        if serializer.is_valid():
+            team = serializer.save()
+            self.create_audit_entry(
+                request=request,
+                organization=team.organization,
+                target_object=team.id,
+                event=AuditLogEntryEvent.TEAM_EDIT,
+                data=team.get_audit_log_data(),
+            )
+
+    def patch(self, request, organization, team):
+        """
+        A SCIM Group PATCH request takes a series of operations to peform on a team.
+        It does them sequentially and if any of them fail no operations should go through.
+        The operations are add members, remove members, replace members, and rename team.
+        """
+        try:
+            with transaction.atomic():
+                for operation in request.data.get("Operations", []):
+                    op = operation["op"].lower()
+                    if op == GroupPatchOps.ADD and operation["path"] == "members":
+                        self._add_members_operation(request, operation, team)
+                    elif op == GroupPatchOps.REMOVE and "members" in operation["path"]:
+                        # the members op contains a filter string like so:
+                        # members[userName eq "baz@sentry.io"]
+                        self._remove_members_operation(request, operation, team)
+                    elif op == GroupPatchOps.REPLACE and not operation.get("path", None):
+                        self._rename_team_opereation(request, operation, team)
+                    elif op == GroupPatchOps.REPLACE and operation["path"] == "members":
+                        # delete all the current team members
+                        # and add the ones in the operation list
+                        with transaction.atomic():
+                            queryset = OrganizationMemberTeam.objects.filter(team_id=team.id)
+                            queryset.delete()
+                            self._add_members_operation(request, operation, team)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist(detail=SCIM_404_USER_RES)
+        # TODO: what other exceptions to catch?
```

Comment:
Do you need to catch IntegrityError too?

Reply:
👍🏼